pkg/asset/machines/azure/azuremachines.go

```diff
@@ -182,7 +182,7 @@ func GenerateMachines(platform *azure.Platform, pool *types.MachinePool, userDat
 			FailureDomain:          ptr.To(mpool.Zones[0]),
 			OSDisk:                 osDisk,
 			AdditionalTags:         tags,
-			AllocatePublicIP:       true,
+			AllocatePublicIP:       false,
```

Comment:
Don't we want a public IP if `publish == External`? At least that's what we do in terraform.

Reply:
This is intentional. The bootstrap can't have a public IP since we are using an outbound LB for the control plane. It breaks the installation from working. We are working on a way around this.

Reply:
So what about a comment explaining this reasoning?

Reply:
I've added a comment explaining things.